File: go/difference-of-squares/difference_of_squares.go
package diffsquares

const testVersion = 1

func Difference(n int) int {
	return SquareOfSums(n) - SumOfSquares(n)
}

func SquareOfSums(n int) int {
	var sum = n * (n + 1) / 2
	return sum * sum
}

func SumOfSquares(n int) int {
	return int(n * (n + 1) * (2*n + 1) / 6)
}

File: swift/difference-of-squares/Sources/DifferenceOfSquares.swift


class Squares {
    let base: Int

    init(_ n: Int) {
        base = n
    }

    var squareOfSums: Int {
        let n = base
        let sum: Int = n * (n + 1) / 2
        return sum * sum
    }

    var sumOfSquares: Int {
        let n = base
        return n * (n + 1) * (2 * n + 1) / 6
    }

    var differenceOfSquares: Int {
        return squareOfSums - sumOfSquares
    }
}